feats/meta.py

```diff
@@ -37,18 +37,30 @@ def __init__(self, obj):
         self.description = obj.__doc__
         self.implementations: Dict[str, Implementation] = {}
         self.annotations = defaultdict(list)
-        for key in dir(obj):
-            value = getattr(obj, key)
-            if key.startswith('_') or not callable(value):
-                continue
-            impl = Implementation(value)
-            self.implementations[impl.name] = impl
-            if hasattr(value, '_feats_annotations_'):
-                for annotation in value._feats_annotations_:
-                    self.annotations[annotation].append(impl)
+
+        if callable(obj):
```

Comment:
This doesn't _necessarily_ mean that `obj` is a function. I guess it comes down to how much that matters here.